Styling/README.md

```diff
@@ -0,0 +1,433 @@
+* `regExp() : RegExp`
+* `regExp(pattern : String, [flags : String]) : RegExp`
+
+Calling `regExp()` with no arguments is the same as calling `regExp('(?:)')`.
+
+If specified, `flags` can have any combination of the following values:
+
+* `g` - global match
+* `i` - ignore case
+* `m` - multiline
+* `u` - unicode
+* `y`- sticky
+
+Regular expressions support the functions:
+* `test(string: String) : Boolean` - Tests the specified string for a match.  
+* `exec(string: String) : String` - Executes a search for a match in the specified string. If the search succeeds, it returns the first instance of a captured `String`. If the search fails, it returns `null`
+
+For example:
+```json
+{
+    "Name" : "Building 1"
+}
+```
+
+```
+regExp('a').test('abc') === true
+regExp('a(.)', 'i').exec('Abc') === 'b'
+regExp('Building\s(\d)').exec(${Name}) === '1'
+```
+
+Regular expressions have a `toString` function for explicit (and implicit) conversion to strings in the format `'pattern'`.
+* `toString() : String`
+
+Regular expressions are treated as `NaN` when performing operations with any operator. 
+
+Regular expressions do not expose any other functions or a `prototype` object.
+
+### Conversions
+
+Style expressions follow JavaScript conversion rules.  To minimize unexpected type coercion, `==` and `!=` operators are not supported.
+
+For conversions involving `Color` or `RegExp`, they are treated as JavaScript objects.  For example, `Color` implicitly converts to `NaN` with `>`, `>=`, `<`, and `<=` operators.  In boolean expressions, a `Color` implicitly converts to `true`, e.g., `!!color() === true`.  In string expressions, `Color` implicitly converts to `String` using its `toString` function.
+
+### Variables
+
+Variables are used to retrieve the property values of individual features in a tileset.  Variables are identified using the ES 6 ([ECMAScript 2015](http://www.ecma-international.org/ecma-262/6.0/)) Template Literal syntax, i.e., `${feature.identifier}` or `${feature['identifier']}`, where the identifier is the case-sensitive property name.  `feature` is implicit and can be omitted in most cases.
+
+Variables can be used anywhere a valid expression is accepted, expcet inside other variable identifiers. For example, the following is not allowed:
+```
+${foo[${bar}]}
+```
+
+If a feature does not have a property with specified name, the variable evaluates to `undefined`.  Note that the property may also be `null` if `null` was explicitly stored for a property.
+
+Variables may be any of the supported native JavaScript types:
+* `Boolean`
+* `Null`
+* `Undefined`
+* `Number`
+* `String`
+
+For example:
+```json
+{
+    "enabled" : true,
+    "description" : null,
+    "order" : 1,
+    "name" : "Feature name"
+}
+```
+
+```
+${enabled} === true
+${description} === null
+${order} === 1
+${name} === 'Feature name'
+```
+
+Variables can be used to constructor colors, for example:
```

Comment:
"to constructor colors": is there a word missing here? Should it be "as constructor colors"?